File: 02 Happy Number/Happy Number.cpp
class Solution
{
public:
    bool isHappy(int n)
    {
        map<int, int> mp;
        bool flag;
        while (1)
        {
            int sum = 0;
            while (n > 0)
            {
                int temp = n % 10;
                sum += (temp * temp);
                n /= 10;
            }
            n = sum;
            if (mp[n] == 1)
            {
                flag = false;
                break;
            }
            if (n == 1)
            {
                flag = true;
                break;
            }
            mp[n]++;
        }
        if (flag)
            return true;
        else
            return false;
    }
};

File: 02 Happy Number/Happy Number.js
/**
 * @param {number} n
 * @return {boolean}
 */
var isHappy = function (n) {
    let set = new Set();
    while (1) {
      let sum = 0;
      while (n > 0) {
        let rem = n % 10;
        sum += rem * rem;
        n = Math.floor(n / 10);
      }
      if (set.has(sum)) {
        flag = 1;
        break;
      } else if (sum === 1) return true;
      else set.add(sum);
      n = sum;
    }
    return false;
  };
  